app/models/certificate.rb

```diff
@@ -214,6 +214,6 @@ def token_for_invalid_certificate(extra)
     reason = validate_cert
 
     Rails.logger.warn("Certificate invalid: #{reason}")
-    TokenService.box(extra.merge(error: "certificate.#{reason}"))
+    TokenService.box(extra.merge(error: "certificate.#{reason}|#{key_id}"))
```

Comment:
Would it be simpler to add a new key to the data/box?

ex:
```
TokenService.box(extra.merge(
  error: "certificate.#{reason}",
  key_id: key_id
)
```

Reply:
No because in IDP the error is logged in the analytics but the key would not be.

Reply:
Right, but rather than explicitly stripping extra stuff on the end of the `error` key, wouldn't it be easier to optionally include this new key?

Reply:
It will definitely not be a one line change in each like it is now.  However if you want me to change it I will.

Reply:
I'm just trying to understand the use cases? Like it seems like `key_id` could be useful in more places, and then it would be easier to access it as its own key in the dictionary...so yes please let's do that